src/dwc_mapping.Rmd

```diff
@@ -348,6 +348,55 @@ write_csv(taxon, here("data", "processed", "taxon.csv"), na = "")
 ```
 
 
+```{r}
+
+```
+
+
+# Create source citation function
+
+All records in the extensions will be documented with a source, i.e. the link to the specific checklist taxon that was used for the specific piece of information. We build this function here first.
+
+```{r}
+build_source_citation <- function(extension, datasetKey_extension){
+  
+  # Select the citation of the source checlist: 
+  
+  checklist_citation <- 
+    checklists %>% 
+    select(datasetKey, citation)
+  
+  # Join the extension and source_citation by their datasetKey:
+  
+  extension <-
+    extension %>% 
+    left_join(
+      checklist_citation,
```

Comment:
I would use `select(checklists, datasetKey, citation)` here and remove the step before (lines 363 to 367).